File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int,int> H(0);
        for (int i = 0; i<n; i++){
            if(H[nums[i]] != 0)
                return true;
            H[nums[i]]++;
        }
        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = []
        pre = 1
        pos = 1
        for i,n in enumerate(nums):
            ans.append(pre)
            pre *= n
        
        for i in range(len(nums)-1,-1,-1):
            ans[i] *= pos
            pos *= nums[i]
            
        return ans
